src/storage/src/hummock/store/version.rs

```diff
@@ -235,3 +264,368 @@ impl HummockReadVersion {
         self.staging.sst.clear();
     }
 }
+
+pub fn read_filter_for_batch(
+    epoch: HummockEpoch, // for check
+    table_id: TableId,
+    key_range: &(Bound<Vec<u8>>, Bound<Vec<u8>>),
+    read_version_vec: Vec<Arc<RwLock<HummockReadVersion>>>,
+) -> StorageResult<(Vec<ImmutableMemtable>, Vec<SstableInfo>, CommittedVersion)> {
+    assert!(!read_version_vec.is_empty());
+    let read_version_guard_vec = read_version_vec
+        .iter()
+        .map(|read_version| read_version.read())
+        .collect_vec();
+    let mut imm_vec = Vec::default();
+    let mut sst_vec = Vec::default();
+    // to get max_mce with lock_guard to avoid losing committed_data since the read_version
+    // update is asynchronous
+    let (lastst_committed_version, max_mce) = {
```

Comment:
IIUC, CommittedVersion will be a pruned HummockVersion. If there are multiple CommittedVersions provided, we should read a union of all of them instead of the one with largest MCE.

However, this makes me think of one thing: is it even possible to have mulitple shards (multiple CommittedVersions) to read in a single read issued by batch executor?

Reply:
A single read issued to storage by batch executor touches a single vnode so only one shard will be touched:
- iter: https://github.com/risingwavelabs/risingwave/blob/d0772a473ecbadaaf3aeea0c67d515292b715cbc/src/storage/src/table/batch_table/storage_table.rs#L318
- get: https://github.com/risingwavelabs/risingwave/blob/d0772a473ecbadaaf3aeea0c67d515292b715cbc/src/storage/src/table/batch_table/storage_table.rs#L251

Reply:
This may change the way we optimize for batch read

Reply:
1 . we haven't yet pruned the CommittedVersion held by each ReadVersions, so we can read one of them with largest MCE.  After pruning , we should read a union of all of them instead of the one with largest MCE.
2. According to your comment, `A single read issued to storage by batch executor touches a single vnode so only one shard will be touched`. So for BatchQuery we can also locate a specific Shard and read only one ReadVersion, the logic here can be simplified. (It depends on how to locate and prune)